mllib/src/main/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifier.scala

```diff
@@ -65,7 +66,42 @@ private[ml] trait MultilayerPerceptronParams extends PredictorParams
   /** @group getParam */
   final def getBlockSize: Int = $(blockSize)
 
-  setDefault(maxIter -> 100, tol -> 1e-4, layers -> Array(1, 1), blockSize -> 128)
+  /**
+   * Optimizer setup.
+   * @group expertParam
+   */
+  final val optimizer: Param[String] = new Param[String](this, "optimizer",
+    " Allows setting the optimizer: minibatch gradient descent (GD) or LBFGS. " +
+      " The latter is recommended one. ",
+    ParamValidators.inArray[String](Array("GD", "LBFGS")))
+
+  /** @group getParam */
+  final def getOptimizer: String = $(optimizer)
+
+  /**
+   * Learning rate.
+   * @group expertParam
+   */
+  final val learningRate: DoubleParam = new DoubleParam(this, "learning rate",
+    " Sets the learning rate for gradient descent optimizer ",
+    ParamValidators.inRange(0, 1))
+
+  /** @group getParam */
+  final def getLearningRate: Double = $(learningRate)
+
+  /**
+   * Model weights. Can be returned either after training or after explicit setting
+   * @group expertParam
+   */
+  final val weights: Param[Vector] = new Param[Vector](this, "weights",
```

Comment:
Is it okay to change it to `coefficients`? We did this for linear models to avoid confusion with instance weights.